commit c09cc58692f8a6ae341e955c5712884f920b84ab
Author: Greg Kroah-Hartman <gregkh@suse.de>
Date:   Fri Oct 8 12:07:33 2010 -0700

    Staging: brcm80211: hnddma.c: fix compiler warning
    
    Change the field types of dma_info_t to make comparing values easier
    (and correct.)  No need to keep rxbufsize as a u16, it can be an
    unsigned int to make things easier.
    
    Cc: Brett Rudley <brudley@broadcom.com>
    Cc: Henry Ptasinski <henryp@broadcom.com>
    Cc: Nohee Ko <noheek@broadcom.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

--- a/drivers/staging/brcm80211/util/hnddma.c
+++ b/drivers/staging/brcm80211/util/hnddma.c
@@ -128,7 +128,7 @@ typedef struct dma_info {
 	u32 rcvptrbase;	/* Base for ptr reg when using unaligned descriptors */
 
 	/* tunables */
-	u16 rxbufsize;	/* rx buffer size in bytes,
+	unsigned int rxbufsize;	/* rx buffer size in bytes,
 				 * not including the extra headroom
 				 */
 	uint rxextrahdrroom;	/* extra rx headroom, reverseved to assist upper stack
@@ -138,7 +138,7 @@ typedef struct dma_info {
 				 *  Some dongle driver may not need it.
 				 */
 	uint nrxpost;		/* # rx buffers to keep posted */
-	uint rxoffset;		/* rxcontrol offset */
+	unsigned int rxoffset;	/* rxcontrol offset */
 	uint ddoffsetlow;	/* add to get dma address of descriptor ring, low 32 bits */
 	uint ddoffsethigh;	/*   high 32 bits */
 	uint dataoffsetlow;	/* add to get dma address of data buffer, low 32 bits */
